### PR TITLE
258951: Business Customer Applications GraphQL Resolvers, Mocks

### PR DESCRIPTION
**Pull request by hadidefra**

### CODE REVIEW

File: app/graphql/context.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 export function context () {
   return {
     dataSources: {
       ruralPaymentsAgencyLandAPI: new RuralPaymentsAgencyLandAPI(),
-      ruralPaymentsPortalApi: new RuralPaymentsPortalApi()
+      ruralPaymentsPortalApi: new RuralPaymentsPortalApi(),
+      api: {
+        organisation: {
+          applications: new OrganisationApplicationsDataSourceAPI()
+        }
```
[the remark this answers is not in the document]
Comment (jack-mccabe-defra) [reply app/graphql/context.js:13]: structure doesn't match the others and I don't think it should be a separate data source

File: app/graphql/types/business/applications.gql [hunk 2]
```diff
@@ -0,0 +1,30 @@
+type BusinessApplication {
+    applicationId: ID
```
Comment (jack-mccabe-defra) [inline app/graphql/types/business/applications.gql:2]: This data does not look like what is being asked for, please check the spreadsheet

File: app/graphql/context.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (hadidefra) [reply app/graphql/context.js:13]: @dtaylor7 I could only see few comments from Jack on closed PR. @jack-mccabe-defra shall we pair up to discuss this?

File: app/graphql/types/business/applications.gql (hunk 2 above)
Comment (hadidefra) [reply app/graphql/types/business/applications.gql:2]: I created this based on what @dtaylor7 given on slack channel, I will ignore that endpoint and match the fields with spreadsheet now

File: app/data-sources/ogranisation/applications.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { RESTDataSource } from '@apollo/datasource-rest'
+
+export class OrganisationApplicationsDataSourceAPI extends RESTDataSource {
```
Comment (dtaylor7) [inline app/data-sources/ogranisation/applications.js:3]: this should be part of `app/data-sources/rural-payments-portal/RuralPaymentsPortalApi.js`

File: app/graphql/context.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dtaylor7) [reply app/graphql/context.js:13]: this is still in a different format

File: mocks/fixtures/organisation.js [hunk 4]
```diff
@@ -46,4 +46,69 @@ export const organisation = {
   charityCommissionRegistrationNumber: null
 }
 
+export const organisationApplications = {
```
Comment (dtaylor7) [inline mocks/fixtures/organisation.js:49]: this should use data from faker, like the rest of the fixtures

File: app/data-sources/ogranisation/applications.js (hunk 3 above)
Comment (jack-mccabe-defra) [reply app/data-sources/ogranisation/applications.js:3]: agreed

File: app/graphql/context.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jack-mccabe-defra) [reply app/graphql/context.js:13]: yep agreed

File: app/transformers/business/application.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  if (!applications) {
+    return applications
+  }
+
+  const result = []
+  for (const response in applications) {
+    result.push({
+      applicationStatus: {
+        id: response.application_id,
+        open: response.application_code,
```
Comment (jack-mccabe-defra) [inline app/transformers/business/application.js:11]: Doesn't seem right a code in this field will not make sense

File: app/transformers/business/application.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        status: response.status,
+        type: response.application_type_de,
```
Comment (jack-mccabe-defra) [inline app/transformers/business/application.js:13]: Presumably mapping provided by James or Ian?

File: app/transformers/business/application.js (hunk 5 above)
Comment (hadidefra) [reply app/transformers/business/application.js:11]: Could you be more specific or add mapping here so I can see the difference, ticket had no context; I mapped this from spreadsheet.

File: mocks/fixtures/organisation.js (hunk 4 above)
Comment (hadidefra) [reply mocks/fixtures/organisation.js:50]: Thanks for this, please see commit: 14e0816

File: app/graphql/context.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (hadidefra) [reply app/graphql/context.js]: I changed it to rural core API. please see commit: 13dc46

File: app/data-sources/ogranisation/applications.js (hunk 3 above)
Comment (hadidefra) [reply app/data-sources/ogranisation/applications.js]: I changed it to rural core API. please see commit: [13dc46](https://github.com/DEFRA/ffc-customer-registry-api/pull/41/commits/13dc46d8f2b1627b4a7bf6cd9b9fce01a86aa658)

File: app/transformers/business/application.js (hunk 6 above)
Comment (hadidefra) [reply app/transformers/business/application.js:13]: I created a Teams chat with Ian, James and you and brought up this topic, please let me know if I could do anything else

File: app/data-sources/rural-payments-agency-org-applications-api.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { RESTDataSource } from '@apollo/datasource-rest'
+
+export class RuralPaymentsAgencyOrganisationApplicationsAPI extends RESTDataSource {
```
Comment (jack-mccabe-defra) [inline app/data-sources/rural-payments-agency-org-applications-api.js:3]: This should jsut be an additional method here: https://github.com/DEFRA/ffc-customer-registry-api/blob/main/app/data-sources/rural-payments-portal/RuralPaymentsPortalApi.js

File: app/data-sources/rural-payments-agency-org-applications-api.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+  baseURL = process.env.RURAL_PAYMENTS_API_URL
```
Comment (jack-mccabe-defra) [inline app/data-sources/rural-payments-agency-org-applications-api.js:4]: You have not added this to the config anywhere

File: app/data-sources/rural-payments-agency-org-applications-api.js (hunk 7 above)
Comment (hadidefra) [reply app/data-sources/rural-payments-agency-org-applications-api.js]: please see commit: [122cb13](https://github.com/DEFRA/ffc-customer-registry-api/pull/41/commits/122cb139c93da9b603f09456d9078840ce2c1878)

File: app/data-sources/ogranisation/applications.js (hunk 3 above)
Comment (hadidefra) [reply app/data-sources/ogranisation/applications.js]: this done, please see final result, and thank you for pairing to speed up review process

**Review by dtaylor7 — DISMISSED** [review]
LGTM

File: app/transformers/rural-payments-portal/applications-cs.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  if (applications.length === 0) {
+    return { applications: [] }
+  }
+
+  const result = []
+  for (const response in applications) {
+    result.push({
+      applicationStatus: {
+        id: response.application_id,
+        open: response.application_code,
```
Comment (jack-mccabe-defra) [inline app/transformers/rural-payments-portal/applications-cs.js:16]: I think we will need to iron out whether this is the right mapping but for now it is fine

File: test/integration/graphql/business.test.js [hunk 10]
```diff
@@ -70,8 +94,84 @@ describe('Query.customer', () => {
...
+                lastMovement
+            }
+        }
+    }
+    `,
+    variableValues: {
+      customerId: '5444918'
+    },
+    schema,
+    contextValue: context()
```
Comment (jack-mccabe-defra) [inline test/integration/graphql/business.test.js:134]: Either I will need to update this or you will depending on my auth PR to use fakeContext

**Review by jack-mccabe-defra — CHANGES_REQUESTED** [review]
@hadidefra Needs a version bump to get a successful build

File: app/graphql/types/business/applications.gql [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+type BusinessApplicationClaims {
+  schemeYear: String
+  type: String
+  status: String
+  lastMovement: String
+}
+
+type BusinessApplication {
+  applicationStatus: BusinessApplicationStatus
+  csClaims: BusinessApplicationClaims
```
Comment (jack-mccabe-defra) [inline app/graphql/types/business/applications.gql:21]: Maybe shouldn't be plural if it is only singular?

**Review by jack-mccabe-defra — DISMISSED** [review]

File: test/integration/graphql/business.test.js [hunk 12]
```diff
@@ -70,8 +95,84 @@ describe('Query.customer', () => {
...
+          id: expect.any(Number),
+          open: null,
+          status: 'Withdrawn',
+          type: expect.any(String),
+          sector: null,
+          year: expect.any(Number),
+          frn: expect.any(Number),
+          office: null
+        },
+        csClaims: {
```
Comment (jack-mccabe-defra) [inline test/integration/graphql/business.test.js:151]: needs updating

File: test/integration/graphql/business.test.js [hunk 13]
```diff
@@ -70,8 +95,84 @@ describe('Query.customer', () => {
...
+        business: JSON.parse(JSON.stringify({
+          ...transformedOrganisation,
+          ...transformedOrganisationCSApplications
+        }))
       }
     })
   })
 })
+
+describe('Query businessApplications', async () => {
```
Comment (jack-mccabe-defra) [inline test/integration/graphql/business.test.js:107]: Can't return async for test

**Review by jack-mccabe-defra — APPROVED** [review]

**Pull request merged**